contrib/slynk-fancy-inspector.lisp

```diff
@@ -6,8 +6,6 @@
 
 (in-package :slynk)
 
-(eval-when (:compile-toplevel :load-toplevel :execute)
-  (slynk-require :slynk-util))
```

Comment:
If we remove this the ASDF-less loading of slynk would probably fail, or be imperiled anyway. Instead what we should do is detect if we're already inside `ASDF/LISP-ACTION:LOAD-OP` and just do a noop in that case. This should be done in the asdf-specific method of `SLYNK:REQUIRE-MODULE` in slynk.lisp. Perhaps @fare could weigh in here for a way to do that.

Reply:
FTR, something like this would already work possibly fine:
```diff
diff --git a/slynk/slynk.lisp b/slynk/slynk.lisp
index dba1ddab..4c66242f 100644
--- a/slynk/slynk.lisp
+++ b/slynk/slynk.lisp
@@ -2919,6 +2919,9 @@ Record compiler notes signalled as `compiler-condition's."
 
 SLY's own `slynk-loader.lisp' is tried first, then ASDF")
 
+(defvar *asdf-load-in-progress* nil
+  "Set to t if inside a \"ASDF:LOAD-SYSTEM\" operation")
+
 (defgeneric require-module (method module)
   (:documentation
    "Use METHOD to load MODULE.
@@ -2927,7 +2930,9 @@ managed to load it.")
   (:method ((method (eql :slynk-loader)) module)
     (funcall (intern "REQUIRE-MODULE" :slynk-loader) module))
   (:method ((method (eql :asdf)) module)
-    (funcall (intern "LOAD-SYSTEM" :asdf) module)))
+    (unless *asdf-load-in-progress*
+      (let ((*asdf-load-in-progress* t))
+        (funcall (intern "LOAD-SYSTEM" :asdf) module)))))
 
 (defun add-to-load-path-1 (path load-path-var)
   (pushnew path (symbol-value load-path-var) :test #'equal))
```

Reply:
Is preserving ASDF-less loading of slynk worth it? If so, wouldn't it be better to write out a dependency graph in a build/loader.lisp file (similar to [jscl](https://github.com/jscl-project/jscl/blob/master/jscl.lisp#L124) instead of calling require inside the files?

Reply:
> Is preserving ASDF-less loading of slynk worth it?

Good question. Maybe not. I've certainly thought about it, Probably I'll keep it for the 1.0 release and then immediately remove it in master. When is the 1.0 release, I don't know :-)

> If so, wouldn't it be better to write out a dependency graph in a build/loader.lisp file (similar to jscl instead of calling require inside the files?

Not sure, no. At least that's not how Emacs does it for example. Ultimately, I'm a fan of preserving information about dependencies in the places where that information is needed, which is the files. But if you can write in 100 lines or less a dependency graph and the loading logic in slynk/slynk-loader.lisp I'm OK. But we'd also need a solution for ASDF-less loading of third party contribs. 

This is why I think the noop solution is the second easiest one, after your solution of ditching ASDF-less loading.